### PR TITLE
Fix : simulate evoked with different sensor types + take into account…

### PR DESCRIPTION
**Pull request by agramfort**
this fixes 2 problems:

- simulate evoked with different sensor types. Symptom was not the noise was not scaled properly for mags. Noise was too strong.
- take into account projs from the cov to make sure that the noise and the signal belong to the same subspace.

thanks @QB3 for the bug report.

### CODE REVIEW

File: mne/simulation/evoked.py [hunk 1]
```diff
@@ -196,9 +198,14 @@ def _generate_noise(info, cov, iir_filter, random_state, n_samples, zi=None):
     rng = check_random_state(random_state)
     c = np.diag(noise_cov.data) if noise_cov['diag'] else noise_cov.data
     mu_channels = np.zeros(len(c))
+    stds = np.sqrt(np.diag(c))
+    stds_inv = np.zeros_like(stds)
+    stds_inv[stds != 0.] = 1. / stds[stds != 0.]
+    c = np.diag(stds_inv).dot(c).dot(np.diag(stds_inv))
     # we almost always get a positive semidefinite warning here, so squash it
     with warnings.catch_warnings(record=True):
```
Comment (larsoner) [inline mne/simulation/evoked.py]: The semidefinites are causing a problem in tests, probably need a `warnings.simplefilter('ignore')` line added below this

**Comment by massich** [comment]
@Qb3 this guy is faling
https://github.com/mne-tools/mne-python/blob/12a85495333e07a80d676dcb0e19a562bd7d380f/mne/beamformer/tests/test_rap_music.py#L115

**Comment by QB3** [comment]
I will check this this evening

**Comment by larsoner** [comment]
Pushed the fix suggested earlier

**Comment by larsoner** [comment]
@agramfort suggested fix pushed, see simpler more unified code + new test that would have failed with the not-positive-semidef error

**Comment by larsoner** [comment]
... and reverted a now-unnecessary residual tolerance change in `test_dipole.py`

**Comment by agramfort** [comment]
thx heaps @larsoner for finishing this one !

@QB3 you can know switch back to MNE master for your experiments.

**Pull request merged**